lib/net/ssh/transport/algorithms.rb

```diff
@@ -368,7 +368,13 @@ def negotiate_algorithms
         def negotiate(algorithm)
           match = self[algorithm].find { |item| @server_data[algorithm].include?(item) }
 
-          raise Net::SSH::Exception, "could not settle on #{algorithm} algorithm" if match.nil?
+          if match.nil?
+            error_message = "could not settle on #{algorithm} algorithm"
+            error { error_message }
+            error { "Server #{algorithm} preferences: #{@server_data[algorithm].join(',')}" }
+            error { "Client #{algorithm} preferences: #{self[algorithm].join(',')}" }
+            raise Net::SSH::Exception, "#{error_message} - see error log for details"
```

Comment:
@dan-jensen thanks much for th PR!

What about adding this to the exception message itself?

Reply:
I would prefer that too, actually, but was concerned about making the exception message too verbose.  But you seem to have the same feeling about it, so I'll push that change shortly.